app/Services/PlanetService.php

```diff
@@ -473,6 +473,9 @@ public function hasResources(Resources $resources): bool
         if (!empty($resources->deuterium->get()) && $this->deuterium()->get() < $resources->deuterium->get()) {
             return false;
         }
+        if (!empty($resources->energy->get()) && $this->energyProduction()->getFormattedLong() < $resources->energy->get()) {
```

Comment:
I'm not sure if this is a typo but I think the `getFormattedLong()` here should be `get()` instead.

Currently the formatted string (e.g. "5.33M") is being compared with the raw number (e.g. 5330000).

Other than this the change itself looks good to me. 👍

Reply:
Thanks! Will adjust, I copied the logic that was showing in the pop up. Leave it with me 👍